File: Data Structures & Algorithms/string-encode-and-decode/submission-0.py
class Solution:

    def encode(self, strs: List[str]) -> str:
        string = ""
        for word in strs:
            string += word 
            string += "#&*"
        return string
    def decode(self, s: str) -> List[str]:
        answer = s.split("#&*")
        answer.pop()
        return answer

File: Data Structures & Algorithms/anagram-groups/submission-0.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        map<map<char,int>,vector<string>> words;
        
        for(auto x : strs){
            map<char,int> word;
            for(auto y : x){
                word[y] += 1;
            }
            words[word].push_back(x);
        }

        vector<vector<string>> answer;

        for(auto x : words){
            answer.push_back(x.second);
        }

        return answer;
    }
};
